### PR TITLE
feat(ci): Move and edit release scripts/workflows

### LINKED ISSUE #13883: Move or copy necessary release scripts into Magma repository
## Tasks
- [ ] Figure out if secrets can be provided securely to Github Action runs. This should be used to give a private artifactory access token to the jobs.
- [ ] Integrate docker promotion script into main magma repo. (Check out if a non-working version is already there?)
- [ ] Integrate helm promotion script into main magma repo. (Check out if a non-working version is already there?)
- [ ] Prepare internal review meeting.

## Acceptance criteria
- Make it possible to promote helm charts and docker containers from within in the magma repo via github action work flows
- Checkin the missing scripts from @ShubhamTatvamasi repo into the main magma repo
- Add required secrets to github actions or as input parameters

## Open Questions:
- Who should be allowed to trigger this workflow?
  - Idea: Use private token as input secret for workflow. -> Only people with access to Artifactory are allowed to use it.

## Implementation details
- Use own deploy repo if we cannot restrict who can execute the action
- Add a variable that defines the artifactory domain in only location in the script, in order to make it possible to migrate to another artifactory.

## Reference:
https://github.com/ShubhamTatvamasi/magma-artifactory-docker-promote
https://github.com/nstng/magma-artifactory-helm-promote

### PR DESCRIPTION
**Pull request by MoritzThomasHuebner**
Closes #13883 

## Summary

- Move release scripts/workflows from:
  - https://github.com/ShubhamTatvamasi/magma-artifactory-docker-promote and
  - https://github.com/nstng/magma-artifactory-helm-promote
- Edit release scripts so they broadly work the same way (e.g. use of environment variables).
- Add magamcore artifactory url variable to make migration to another artifactory easier.

- Configure the workflow so that only the `approvers-ci` codeowners can run it.

## Test Plan

I don't have the necessary permissions for the artifactory to really test it. It would be good to run these scripts in some way to see if they work as expected. However, I do not have the necessary permissions to perform any such tests. I tried to test the workflow team membership checks in my own testing organization. However, I would need a paid tier subscription to use secrets. 

It is also difficult to run `workflow-dispatch` actions if the workflow does not yet exist on master. As far as I can tell, I would have to paste the code temporarily into an existing `workflow-dispatch` workflow.

### CODE REVIEW

**Review by tmdzk — APPROVED** [review]

**Comment by maxhbr** [comment]
The files need to have license headers

**Comment by MoritzThomasHuebner** [comment]
> The files need to have license headers

Thanks, I added the headers now.

File: .github/workflows/docker-promote.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+name: docker promote
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch_tag:
+        description: 'Branch version number'
+        required: true
+        default: '1.8'
```
Comment (sebathomas) [inline .github/workflows/docker-promote.yml]: I don't think we should have defaults for these values. Whoever uses these scripts should take care to choose the right version.

File: .github/workflows/docker-promote.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      release_tag:
+        description: 'Release version number'
+        required: true
+        default: '1.8.0'
```
Comment (sebathomas) [inline .github/workflows/docker-promote.yml]: Same here.

File: .github/workflows/helm-promote.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+name: helm promote
+
+on:
+  workflow_dispatch:
+    inputs:
+      magma_version:
+        description: 'Magma version number'
+        required: true
+        default: '1.8.0'
```
Comment (sebathomas) [inline .github/workflows/helm-promote.yml]: Same here.

**Review by sebathomas — COMMENTED** [review]
Looking good, just some minor comments.

Too bad there's no easy way to test this...

File: .github/workflows/docker-promote.yml (hunk 1 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/docker-promote.yml]: Fixed

File: .github/workflows/docker-promote.yml (hunk 2 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/docker-promote.yml]: Fixed

File: .github/workflows/helm-promote.yml (hunk 3 above)
Comment (MoritzThomasHuebner) [reply .github/workflows/helm-promote.yml]: Fixed

**Review by sebathomas — APPROVED** [review]

**Pull request merged**